tooling/nargo/src/ops/foreign_calls.rs

```diff
@@ -120,9 +120,12 @@ impl DefaultForeignCallExecutor {
         decode_string_value(&fields)
     }
 
-    fn execute_println(foreign_call_inputs: &[ForeignCallParam]) -> Result<(), ForeignCallError> {
+    fn execute_print(foreign_call_inputs: &[ForeignCallParam]) -> Result<(), ForeignCallError> {
         let display_values: PrintableValueDisplay = foreign_call_inputs.try_into()?;
-        println!("{display_values}");
+        let last_input = foreign_call_inputs.last().expect("print expected at least 1 argument");
+        let skip_newline = last_input.unwrap_value().is_zero();
+        print!("{display_values}{}", if skip_newline { "" } else { "\n" });
+
```

Comment:
My last review was wrong. `skip_newline` should be the third to last parameter, not the last. I forgot about the format string and another field constant.
```suggestion

        // The skip_newline parameter is the third to last, behind the format string.
        // Since the type to print is generic, it may take up more than 1 argument
        // so we need to count back from the end here instead of starting from the beginning.
        let skip_newline = &foreign_call_inputs[foreign_call_inputs.len() - 3];
        let skip_newline = skip_newline.unwrap_value().is_zero();

        print!("{display_values}{}", if skip_newline { "" } else { "\n" });
```